drivers/timers/CMakeLists.txt

```diff
@@ -22,6 +22,13 @@
 
 set(SRCS)
 
+# Unless a driver with a more accurate version of up_*delay is enabled, always
+# include the base delay definition (busy-loop) as a minimum-viable product.
+
+if(NOT CONFIG_TIMER_ARCH)
```

Comment:
why not remove the same implementation from arch_timer.c too?

Reply:
The timer implementation actually uses a different definition for the delays; it chooses between a software implementation (the standard one) or a hardware implementation depending on the duration.

Reply:
your implementation could call clock_systime_timespec to achieve the similar behavior and then remove arch_timer.c's.

Reply:
I am not familiar with this function, is this a feature on all architectures? Or why is `arch_timer` generally separate?

Reply:
After looking at this, I think it would need a separate PR and time to think. The `up_udelay` in the timer implementation is dependent on the timer driver lower-half. I'm not certain right now what the best way to replace that with `clock_systime_timespec` is, and it would require a lot more testing.